### ISSUE TITLE
Non existent member not detected on datetime.datetime

### ISSUE THREAD

**Issue by dmmartins** [pylint-dev/pylint#9497]
### Bug description

```python
"""
Pylint test
"""

import datetime

print(datetime.datetime.not_a_member)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```python
--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```

### Expected behavior

I should detect a `no-member` error

### Pylint version

```shell
pylint 3.1.0
astroid 3.1.0
Python 3.11.4 (main, Aug 17 2023, 00:58:58) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

### OS / Environment

macOS 13.6.1

**Comment by Pierre-Sassoulas** [comment]
Fixed in pyliny 3.0
